### PR TITLE
Remove `try_mlflow_log` from autologging & tests

### PR DESCRIPTION
**Pull request by liangz1**
Signed-off-by: Liang Zhang <liang.zhang@databricks.com>

## What changes are proposed in this pull request?

Remove `try_mlflow_log` from autologging & tests

## How is this patch tested?

Existing tests

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [x] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/keras.py [hunk 1]
```diff
@@ -718,7 +717,7 @@ def _log_early_stop_callback_params(callback):
                 "baseline": callback.baseline,
                 "restore_best_weights": callback.restore_best_weights,
             }
-            try_mlflow_log(mlflow.log_params, earlystopping_params)
+            mlflow.log_param(earlystopping_params)
```
Comment (harupy) [inline mlflow/keras.py:720]:
```suggestion
            mlflow.log_params(earlystopping_params)
```

**Comment by harupy** [comment]
To fix lint failures for `black`,  you can just run:

```
pip install black==19.10b0
black .
```

**Comment by harupy** [comment]
Could you merge the latest master branch? That should solve the CI failures.

File: mlflow/keras.py [hunk 2]
```diff
@@ -645,24 +644,22 @@ class __MLflowKerasCallback(keras.callbacks.Callback, metaclass=ExceptionSafeCla
             """
 
             def on_train_begin(self, logs=None):  # pylint: disable=unused-argument
-                try_mlflow_log(mlflow.log_param, "num_layers", len(self.model.layers))
-                try_mlflow_log(
-                    mlflow.log_param, "optimizer_name", type(self.model.optimizer).__name__
-                )
+                mlflow.log_params("num_layers", len(self.model.layers))
+                mlflow.log_params("optimizer_name", type(self.model.optimizer).__name__)
```
Comment (harupy) [inline mlflow/keras.py:648]:
```suggestion
                mlflow.log_param("num_layers", len(self.model.layers))
                mlflow.log_param("optimizer_name", type(self.model.optimizer).__name__)
```

File: mlflow/keras.py [hunk 3]
```diff
@@ -645,24 +644,22 @@ class __MLflowKerasCallback(keras.callbacks.Callback, metaclass=ExceptionSafeCla
...
                 if hasattr(self.model.optimizer, "lr"):
                     lr = (
                         self.model.optimizer.lr
                         if type(self.model.optimizer.lr) is float
                         else keras.backend.eval(self.model.optimizer.lr)
                     )
-                    try_mlflow_log(mlflow.log_param, "learning_rate", lr)
+                    mlflow.log_params("learning_rate", lr)
```
Comment (harupy) [inline mlflow/keras.py:655]:
```suggestion
                    mlflow.log_param("learning_rate", lr)
```

File: mlflow/keras.py [hunk 4]
```diff
@@ -645,24 +644,22 @@ class __MLflowKerasCallback(keras.callbacks.Callback, metaclass=ExceptionSafeCla
...
                 if hasattr(self.model.optimizer, "epsilon"):
                     epsilon = (
                         self.model.optimizer.epsilon
                         if type(self.model.optimizer.epsilon) is float
                         else keras.backend.eval(self.model.optimizer.epsilon)
                     )
-                    try_mlflow_log(mlflow.log_param, "epsilon", epsilon)
+                    mlflow.log_params("epsilon", epsilon)
```
Comment (harupy) [inline mlflow/keras.py:662]:
```suggestion
                    mlflow.log_param("epsilon", epsilon)
```

File: mlflow/utils/autologging_utils/safety.py [hunk 5]
```diff
@@ -26,19 +26,6 @@
...
-    """
-    Catch exceptions and log a warning to avoid autolog throwing.
-    """
-    try:
-        return fn(*args, **kwargs)
-    except Exception as e:
-        if is_testing():
-            raise
-        else:
-            warnings.warn("Logging to MLflow failed: " + str(e), stacklevel=2)
```
Comment (harupy) [inline mlflow/utils/autologging_utils/safety.py:39]: Can we fix this lint error by removing `import warnings` in this file?

https://github.com/mlflow/mlflow/runs/4070952017?check_suite_focus=true#step:6:26

```
mlflow/utils/autologging_utils/safety.py (7,0): [W0611 unused-import] Unused import warnings
```

File: examples/pytorch/MNIST/mnist_autolog_example.py [hunk 6]
```diff
@@ -16,7 +16,7 @@
 from pytorch_lightning.callbacks.early_stopping import EarlyStopping
 from pytorch_lightning.callbacks import ModelCheckpoint
 from pytorch_lightning.callbacks import LearningRateMonitor
-from pytorch_lightning.metrics.functional import accuracy
+from torchmetrics.functional import accuracy
```
Comment (harupy) [inline examples/pytorch/MNIST/mnist_autolog_example.py:19]: Can we revert this change since it's unrelated to this PR?

Yesterday I filed https://github.com/mlflow/mlflow/pull/4976 and fixed the issue in the example check.

Comment (harupy) [reply examples/pytorch/MNIST/mnist_autolog_example.py:19]: If #4976 didn't actually solve the issue and we need another fix, we can create a separate PR.

**Comment by harupy** [comment]
I filed https://github.com/mlflow/mlflow/pull/4984 to fix the following issues:
- `Cross version tests / test (tensorflow / 2.6.0 / autologging)`.
- `Cross version tests / test (tensorflow / 2.6.0 / models)`.

File: examples/pytorch/MNIST/mnist_autolog_example.py (hunk 6 above)
Comment (harupy) [reply examples/pytorch/MNIST/mnist_autolog_example.py:19]: @liangz1 On a second thought, can we revert the changes #4976 made, and apply your fix in a separate PR?

Comment (liangz1) [reply examples/pytorch/MNIST/mnist_autolog_example.py]: Sure, I created https://github.com/mlflow/mlflow/pull/4989 for the fix and revert.

File: mlflow/utils/autologging_utils/__init__.py [hunk 7]
```diff
@@ -224,7 +223,7 @@ def _timed_log_batch(self):
             for i in range(0, len(self.data), MAX_METRICS_PER_BATCH)
         ]
         for metrics_slice in metrics_slices:
-            try_mlflow_log(self.client.log_batch, run_id=current_run_id, metrics=metrics_slice)
+            self.client.log_batch(run_id=current_run_id, metrics=metrics_slice)
```
Comment (WeichenXu123) [inline mlflow/utils/autologging_utils/__init__.py:226]: The test `test_batch_metrics_logger_continues_if_log_batch_fails` failed because of this line change behavior.
Here we still need `try_mlflow_log` logic to keep the behavior which the test name describes.

**Comment by WeichenXu123** [comment]
The test `test_import_tensorflow_with_fluent_autolog_enables_tf_autologging` and `test_import_tf_keras_with_fluent_autolog_enables_tf_autologging` failure on tensorflow 2.5.2 cross-test failure is an irrelative issue.
I will create another PR to fix it.

**Comment by WeichenXu123** [comment]
```
    def exception_safe_function(function):
        """
        Wraps the specified function with broad exception handling to guard
        against unexpected errors during autologging.
        """
        if is_testing():
>           setattr(function, _ATTRIBUTE_EXCEPTION_SAFE, True)
E           AttributeError: 'method' object has no attribute 'exception_safe'

function   = <bound method MlflowClient.log_batch of <mlflow.tracking.client.MlflowClient object at 0x7f1b5c1ba1d0>>

mlflow/utils/autologging_utils/safety.py:39: AttributeError
```

We cannot set attribute into bound method. The line `setattr(function, _ATTRIBUTE_EXCEPTION_SAFE, True)` cause new test error.
